actions/detect-exposed-workflow-secrets/exposed_secrets_detection/src/cli.py

```diff
@@ -0,0 +1,179 @@
+# SPDX-License-Identifier: Apache-2.0
+
+# Standard
+import argparse
+import sys
+
+# Local
+from .custom_exceptions import (
+    GitWorkflowFilesNotFoundError,
+    ExposedSecretsError,
+)
+from .utils import (
+    find_all_yaml_files_in_directory,
+    find_exposed_github_secrets_in_env_block,
+    find_exposed_github_secrets_in_job,
+    get_workflow_trigger_conditions,
+    is_git_workflow_file,
+    load_workflow_file,
+    workflow_auto_triggers_on_pull_request,
+)
+
+
+def create_parser() -> argparse.ArgumentParser:
+    """
+    Generates a parser for the CLI.
+
+    Returns
+    -------
+    parser: argparse.ArgumentParser
+    """
+    parser = argparse.ArgumentParser(
+        prog="Exposed-Secrets-Detection",
+        description="Attempts to find any potentially exposed secrets, and reports them",
+    )
+
+    # User can pass in a directory or file, hence they both have "required=False", but one must be chosen
+    parser.add_argument(
+        "-d",
+        "--dir",
+        type=str,
+        help="Directory to read Git workflow files from",
+        required=False,
+    )
+    parser.add_argument(
+        "-f",
+        "--file",
+        type=str,
+        help="Path to the Git workflow file to read",
+        required=False,
+    )
+
+    return parser
+
+
+def run_validation(args: dict):
+    """
+    Runs validation steps to detect potentially-exposed secrets. In the process, this function also validates
+    that the user has provided at least "--dir" or "--file" as inputs
+
+    Inputs
+    ------
+    args: dict
+        Dictionary of the user's arguments
+    """
+    read_dir = args.get("dir")
+    file_path = args.get("file")
+
+    if not read_dir and not file_path:
+        print(
+            "This tool requires either: a path to a valid Git workflow file or a directory that contains Git workflow files. Detected neither."
+        )
+        sys.exit(1)
+
+    if read_dir and file_path:
+        raise NotImplementedError(
+            "This tool currently does not support both --dir and --file being passed in simultaneously."
+        )
+
+    if read_dir:
+        print(
+            f"Finding GitHub workflow files under {read_dir} and its sub-directories ..."
+        )
+        git_workflow_files = {}
+        yaml_files = find_all_yaml_files_in_directory(read_dir)
+
+        # For each YAML file we've found, store it in a dictionary for later parsing. We need the yaml file *path* as the
+        # key so we can provide actionable error messages if an exception is caught.
+        workflow_files_found = 0
+        for yaml_file in yaml_files:
+            loaded_file = load_workflow_file(yaml_file)
+
+            # Only parse the file if we've confirmed it's a valid Git workflow file and not a random YAML
+            if is_git_workflow_file(loaded_file):
+                git_workflow_files.update({yaml_file: loaded_file})
```

Comment:
Another update case, did you want to change this one too?

Reply:
Thanks. Updated.